src/Geometry/PUMLReader.cpp

```diff
@@ -239,32 +310,41 @@ void PUMLReader::partition(PUML::TETPUML& puml,
 
   auto newPartition = partitioner->partition(graph, target);
 
-  puml.addDataArray(ltsWeights->clusterIds().data(), PUML::CELL, {});
-  puml.addDataArray(ltsWeights->timesteps().data(), PUML::CELL, {});
+  meshGeometry.addDataArray(ltsWeights->clusterIds().data(), PUML::CELL, {});
+  meshGeometry.addDataArray(ltsWeights->timesteps().data(), PUML::CELL, {});
 
-  puml.partition(newPartition.data());
+  meshGeometry.partition(newPartition.data());
+  if (&meshTopology != &meshGeometry) {
+    meshTopology.partition(newPartition.data());
+  }
 }
 
-void PUMLReader::generatePUML(PUML::TETPUML& puml) {
+void PUMLReader::generatePUML(PumlMesh& meshTopology, PumlMesh& meshGeometry) {
   SCOREP_USER_REGION("PUMLReader_generate", SCOREP_USER_REGION_TYPE_FUNCTION);
 
-  puml.generateMesh();
+  if (&meshTopology != &meshGeometry) {
+    meshTopology.generateMesh();
+  }
+  meshGeometry.generateMesh();
 }
 
-void PUMLReader::getMesh(const PUML::TETPUML& puml) {
+void PUMLReader::getMesh(const PumlMesh& meshTopology, const PumlMesh& meshGeometry) {
   SCOREP_USER_REGION("PUMLReader_getmesh", SCOREP_USER_REGION_TYPE_FUNCTION);
 
   const int rank = MPI::mpi.rank();
 
-  const std::vector<PUML::TETPUML::cell_t>& cells = puml.cells();
-  const std::vector<PUML::TETPUML::face_t>& faces = puml.faces();
-  const std::vector<PUML::TETPUML::vertex_t>& vertices = puml.vertices();
+  const std::vector<PumlMesh::cell_t>& cells = meshTopology.cells();
```

Comment:
so what would be the difference between meshTopology.cells() and meshGeometry.cells().
I m not sure I can follow.